### PR TITLE
test: data pipeline implementation interaction

### PR DESCRIPTION
**Pull request by mrehan27**
helps: https://github.com/customerio/issues/issues/11965

###  Changes

- Fixed tests for `DataPipelineImplementation`
- Added tests for verifying `DataPipeline` communication with `analytics`

### CODE REVIEW

File: Sources/Common/Communication/EventBusHandler.swift [hunk 1]
```diff
@@ -1,11 +1,19 @@
 import Combine
 import Foundation
 
+public protocol EventBusHandler {
```
Comment (mrehan27) [inline Sources/Common/Communication/EventBusHandler.swift:4]: Added interface so calls to EventBus can be tested

File: Sources/DataPipeline/DataPipelineImplementation.swift
Comment (mrehan27) [inline Sources/DataPipeline/DataPipelineImplementation.swift:1]: The file contains changes to fix bugs/behavior changes identified while fixing and running these tests

File: Tests/Shared/Mocks/EventBusHandler.swift [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import CioInternalCommon
+
+/// Mock of the EventBusHandler class, designed to mimic AutoMockable
+/// Once the class is generated using AutoMockable, it should seamlessly replace the current implementation without any issues
+public class EventBusHandlerMock: EventBusHandler, Mock {
```
Comment (mrehan27) [inline Tests/Shared/Mocks/EventBusHandler.swift:5]: Due to generics, I couldn't generate it using `AutoMockable`, but I tried to keep the implementation similar to it. If we can generate this using `AutoMockable`, that should hopefully work without making any change in tests.

File: Tests/Tracking/migration/DataPipelineMigrationAssistantTests.swift [hunk 3]
```diff
@@ -0,0 +1,76 @@
+@testable import CioDataPipelines
+@testable import CioInternalCommon
+import Foundation
+@testable import Segment
+import SharedTests
+import XCTest
+
+class DataPipelineMigrationAssistantTests: UnitTest {
```
Comment (mrehan27) [inline Tests/Tracking/migration/DataPipelineMigrationAssistantTests.swift:8]: I just moved them from [`CustomerIOImplementationTests`](https://github.com/customerio/customerio-ios/blob/main-replica-for-cdp/Tests/DataPipeline/DataPipelineInteractionTests.swift#L413-L452) to a separate file.

File: Sources/DataPipeline/DataPipelineImplementation.swift
Comment (mrehan27) [inline Sources/DataPipeline/DataPipelineImplementation.swift:1]: The file contains changes to fix bugs/behavior changes identified while fixing and running these tests

Comment (mrehan27) [reply Sources/DataPipeline/DataPipelineImplementation.swift:1]: Resolving as shown outdated

Comment (mrehan27) [inline Sources/DataPipeline/DataPipelineImplementation.swift:1]: The file contains changes to fix bugs/behavior changes identified while fixing and running these tests

File: Tests/Shared/Mocks/EventBusHandler.swift (hunk 2 above)
Comment (Shahroz16) [reply Tests/Shared/Mocks/EventBusHandler.swift:5]: i think it's a fair call, using generic here really helps with the casting of events when mapped from map

**Review by Shahroz16 — APPROVED** [review]

File: Tests/Shared/Mocks/EventBusHandler.swift (hunk 2 above)
Comment (mrehan27) [reply Tests/Shared/Mocks/EventBusHandler.swift:5]: Agreed

**Review by ami-aman — APPROVED** [review]

**Pull request merged**